### DISCUSSION TITLE
HP 1650A Logic Analyzer FlashFloppy'd

### DISCUSSION THREAD

**Discussion by phvidston** [keirf/flashfloppy#870] [Show and tell]
I am happy to report that I have my (t)rusty old HP1651A running with FlashFloppy. Another big thanks to Keir for an amazing firmware package. I had no easy way to transfer the uniquely formatted disks, but was successful using "fdio -dup -raw" after analyzing the disk format on a PC with a 300 RPM drive, again using fdio. The HP1650A uses DSDD MFM 500KHz, 600 RPM, 80 cylinder, 5 sectors at 1024 bytes/sector. The first 79 cylinders (0-78) have normal sector numbering of 1-5. The last cylinder has sector numbers 97-101 with copy protection info, so you need to get all the cylinders. I also had to 3D-print a filler strip to account for the shorter height of the GOTEK as compared to the original Sony drive. If anyone is interested I can provide disk images for HP OS v1.02 and v1.10, as well as information on the adapter board and configuration. I'm so happy to have this piece of T.E. I paid so much for new, brought up to date. Cheers, Paul

**Comment by joseluizmartins** [comment]
Hi Paul!

Good news you have make it working! Could you share the image and the STL you used to print the filler? I'm newbie on Gotek, and trying to understand how it works. All help will be very welcome! 

Thank you vey much!


JL

**Comment by phvidston** [comment]
JL, I'm attempting to post the files you requested. First I'll try the STL file of the filler. It's embarrassingly simple but works OK. 
Cheers, Paul
[HP1650Filler.zip](https://github.com/user-attachments/files/16488731/HP1650Filler.zip)

**Comment by phvidston** [comment]
JL,
OK, that seemed to work. Here's my "DSKA0000 OS v110.img" for the 1651A.
Cheers, Paul
[DSKA0000 OS v110.zip](https://github.com/user-attachments/files/16488741/DSKA0000.OS.v110.zip)

**Reply by joseluizmartins** [reply]
Thank you very much! You are SUPER!

I'm newbie here... updated the Firmware as noted on a Gotek and I have some questions.

I use one FF.CFG file as:

interface=shugart
pin02=chg
pin34=rdy
chgrst=pa14

And IMG.CFG:

cyls=80
heads=2
rpm=600
mode=mfm
rate=500
secs=16
bps=256
tracks=0-77
  id=1
tracks=78
  id=17
tracks=79
  id=97

What else I need to do please? 

Thank you very much!


JL

**Comment by joseluizmartins** [comment]
Is it ever necessary some modification on the board or wiring?

Thank you!

**Reply by phvidston** [reply]
Yes, a 34 pin adapter board and an additional wire on the GOTEK board are required. Most importantly, the HP supplies +5 and +12 volts on odd-numbered pins of the 34 pin connector. These must be deleted (since odd-numbered pins are grounded on the GOTEK), and +5V routed to the GOTEK power connector. Also, "DISK CHANGE RESET" (pin 1 on the 34 pin connector from HP) gets re-routed to unused pin 4 of the GOTEK, and then pin 4 of the 34-pin connector on the GOTEK board is jumpered to SWCLK on the programming connector, as documented in the FF.CFG file. Please note that these instructions apply to the original(?) GOTEK boards (only ones I have) so you may need to adapt to newer versions of GOTEK H/W. In this post, I have included a picture of my installation, a file documenting the 34-pin adapter, and a ZIP file containing my FF.CFG and IMG.CFG files.

[current.zip](https://github.com/user-attachments/files/16495554/current.zip)

[HP to FF 34pin cable.txt](https://github.com/user-attachments/files/16495530/HP.to.FF.34pin.cable.txt)

![GOTEK in HP1651 LA](https://github.com/user-attachments/assets/28ae9990-c866-45a3-8708-5f919528eb4a)

**Reply by joseluizmartins** [reply]
THANK YOU VERY MUCH!

I'll follow your instructions and post the results.

Thank you !


JL

**Comment by joseluizmartins** [comment]
Mine is not working... 

Checked the wiring and the files several times, but not success. I even, bought a new Gotek, but don't know if it is original or not.
It starts reading, goes from T0 to T79 in less them 1 second, back to T78 and the 1650A beeps.
Here a picture of my setup. 

I know in USA this machines are very cheap, but I live in Brazil and it costs about US$ 1,000 a used one down here!


All help or suggestion is very wellcomed!

![WhatsApp Image 2024-08-11 at 21 02 35](https://github.com/user-attachments/assets/4ce44771-67d3-4e88-8584-c3a470cd2889)

**Comment by phvidston** [comment]
Hello JL. Sorry that you are having problems. Here's my thoughts on what may be happening:

First our configurations are different. I noticed that in your IMG.CFG file shown above, you incorrectly have:
tracks=78
id=17
This should not be there, and may be your problem. 

In your FF.CFG file, Pin Change should read:
pin02=nchg

Also, I have "chgrst=pa14" commented out in my configuration.
I would suggest that you use my two CFG files as-is, and see if that gets you running. If not, see my second and third thoughts below.

Second, it looks like you are using a newer board, and I have no idea if it is compatible with the mod on the board. I'm including pictures of my board (jumper wire is on the bottom of board).

Third, I have only tested this on a 1651A, not a 1650A, so I don't know if the disk image will work with that model.

Finally, here's a directory listing of what I have on my USB drive used with my 1651A:
10/13/2005  11:32 PM           819,200 DSKA0000 OS v110.1650A.IMG
10/13/2005  11:32 PM           819,200 DSKA0001 OS v110.1650A.IMG
10/13/2005  11:32 PM           819,200 DSKA0002 OS v110.1650A.IMG
10/13/2005  11:32 PM           819,200 DSKA0003 OS v110.1650A.IMG
10/13/2005  11:32 PM           819,200 DSKA0004 OS v110.1650A.IMG
10/13/2005  11:32 PM           819,200 DSKA0005 OS v110.1650A.IMG
01/01/2002  12:56 AM           819,200 DSKA0006 OS v1.02.1650A.IMG
02/05/2024  06:43 PM           819,200 DSKA0007 OS v110.1650A.IMG
01/01/2002  12:56 AM           819,200 DSKA0008 OS v1.02.1650A.IMG
02/03/2024  02:16 PM               749 FF.CFG
01/01/2019  12:00 AM                 0 IMAGE_A.CFG
02/03/2024  08:03 AM               829 IMG.CFG

![GOTEK Board](https://github.com/user-attachments/assets/f3eaeb6f-8840-4f51-bc67-5930b62540e8)
![GOTEK Board bottom](https://github.com/user-attachments/assets/de302e8d-bbfc-4f6f-95e8-e06973fc6bd4)

**Reply by joseluizmartins** [reply]
Thank you very much for your quick response.

Forgot to mention, I'm using your both CFG and IMG files as your image file too.

I'll read carefully all your instructions.


Thank you very much!

**Reply by joseluizmartins** [reply]
Hi Paul! Thanks again for your answer!

> First our configurations are different. I noticed that in your IMG.CFG file shown above, you incorrectly have: tracks=78 id=17 This should not be there, and may be your problem.
> 
> In your FF.CFG file, Pin Change should read: pin02=nchg

As said before, I'm using your both FF and CFG files, thanks for sharing!

After a close reading of all your documentation I found this at FF.CFG:

# Plug1Pin6(34 pin conn) to PrgPort pin6 (row closest to pwr conn, 3rd from right),
# on bottom of board.

Did you wired the Plug Pin #6 from 34 pin conn from Gotek to PrgPort pin #6?


Thank you very much!


JL

**Reply by phvidston** [reply]
Hi JL,
Sorry for the confusion. The comment in FF.CFG is incorrect (Wiring diagram is correct). It is Plug1Pin4. See Hardware Mods in the Flash Floppy Wiki. That being said, I haven't enabled the Disk Change Reset function in software, as it is commented out in FF.CFG. It seems that the HP 1651A works with Disk Change being cleared by Step (Flash Floppy default). I did the hardware first, and then worked with the config files and found I did not need to enable "chgrst=pa14" to get the 1651A working. Some versions of the GOTEK board do not support Disk Change Reset in hardware (e.g. SFRKC30.AT2), so it may not be appropriate on your board. I would try without this mod, just to see if connecting to that signal is causing your problem.

A BIG CAUTION is that HP manufactured the ribbon cable backwards, such that the stripe is not on pin 1, but is on pin 34. It appears you know this, but I flag this for other HP1651A users. Your drive would not have powered up if cable was reversed.

If you are still having problems, and all the hardware seems correct, you might consider that your Dupont jumpers might be causing noise on the data lines. A quick test might be to twist a ground wire with READ DATA and have that ground wire grounded at the GOTEK and the HP ribbon cable. Beyond that, it may just be that your newer GOTEK board needs additional configuration. Keir may need to provide an opinion on that.

-Best regards, Paul

**Comment by joseluizmartins** [comment]
Hi Paul!

Thank you very much for your effort and time answering my questions!

JL
